From f338be367117ccc9ebc0f502ab222c3c4318ba82 Mon Sep 17 00:00:00 2001
From: Icenowy Zheng <zhengxingda@iscas.ac.cn>
Date: Thu, 7 May 2026 16:17:07 +0800
Subject: [PATCH 095/344] BACKPORT: FROMLIST: riscv: dts: thead: lpi4a: sort
 nodes

Although "D" and "H" are earlier in the alphabet than "P", the DPU and
HDMI nodes were added after PADCTRL node in the Lichee Pi 4A device tree.

Sort the nodes in this device tree.

Signed-off-by: Icenowy Zheng <zhengxingda@iscas.ac.cn>
Link: https://lore.kernel.org/r/20260507081710.4090814-10-zhengxingda@iscas.ac.cn
Signed-off-by: Han Gao <gaohan@iscas.ac.cn>

[ Mingcong Bai: Resolved a minor post-7.1 merge conflict in
  arch/riscv/boot/dts/thead/th1520-lichee-pi-4a.dts ]

Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 arch/riscv/boot/dts/thead/th1520-lichee-pi-4a.dts | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/arch/riscv/boot/dts/thead/th1520-lichee-pi-4a.dts b/arch/riscv/boot/dts/thead/th1520-lichee-pi-4a.dts
index b659ffd7eb87c..25f5c9d3f0d94 100644
--- a/arch/riscv/boot/dts/thead/th1520-lichee-pi-4a.dts
+++ b/arch/riscv/boot/dts/thead/th1520-lichee-pi-4a.dts
@@ -252,6 +252,20 @@ i2c-pins {
 	};
 };
 
+&dpu {
+	status = "okay";
+};
+
+&hdmi {
+	status = "okay";
+};
+
+&hdmi_out_port {
+	hdmi_out_con: endpoint {
+		remote-endpoint = <&hdmi_con_in>;
+	};
+};
+
 &padctrl0_apsys {
 	fan_pins: fan-0 {
 		pwm1-pins {
-- 
2.52.0

